models/models.go

```diff
@@ -0,0 +1,41 @@
+package models
+
+import (
+	"errors"
+	"time"
+)
+
+type Book struct {
+	ID       string     `json:"id" bson:"_id"`
+	Title    string     `json:"title" bson:"title"`
+	Author   string     `json:"author" bson:"author"`
+	Synopsis string     `json:"synopsis,omitempty" bson:"synopsis"`
```

Comment:
As Rhys mentioned we should probably add the extra value `omitempty` to the bson tags that are optional fields, like so:

```
	Synopsis string     `json:"synopsis,omitempty" bson:"synopsis,omitempty"`
	Links    *Link      `json:"links,omitempty"    bson:"links,omitempty"`
	History  []Checkout `json:"history,omitempty"  bson:"history,omitempty"`
```